src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -503,9 +505,9 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
             var isNotNullable = underlyingNullableType is null;
 
             var nonNullableParameterType = underlyingNullableType ?? parameter.ParameterType;
-            var tryParseMethod = TryParseMethodCache.FindTryParseMethod(nonNullableParameterType);
+            var tryParseMethodCall = TryParseMethodCache.FindTryParseMethodCall(parameter);
```

Comment:
We should continue using `nonNullableParameterType` here. I don't see why `FindTryParseMethodCall` needs the whole ParameterInfo just to extract the non-nullable parameter type again.

Nit: Keep the old `TryParseMethodCache.FindTryParseMethod` name. You can even keep the old `tryParseMethod` variable name. It's just a different way to represent the method. You don't get the actual method *call* Expression into you invoke the `Func` anyway.